### DISCUSSION TITLE
TS0501B Remember Last State

### DISCUSSION THREAD

**Discussion by ster1um** [Koenkk/zigbee2mqtt#9680] [Devices]
Hello,
This device (tuya TS0501B) don't remember the last state after power loss. 
Before return the power remains on regardless of the last state.

there is a chance to remember the last state?

```
{
    "brightness": 254,
    "color_mode": "xy",
    "linkquality": 141,
    "state": "ON",
    "device": {
        "applicationVersion": 91,
        "dateCode": "",
        "friendlyName": "0xa4c1380803e24591",
        "hardwareVersion": 1,
        "ieeeAddr": "0xa4c1380803e24591",
        "manufacturerID": 4417,
        "manufacturerName": "_TZ3210_9q49basr",
        "model": "TS0501B",
        "networkAddress": 49203,
        "powerSource": "Mains (single phase)",
        "stackVersion": 0,
        "type": "Router",
        "zclVersion": 3
    }
}
```

**Comment by ster1um** [comment]
I try to find any solution for this behaviour but no way for remember last state😥

**Comment by ster1um** [comment]
anyone have solution for this behaviour?

**Comment by bedson87** [comment]
Nothing has changed, the device still does not save its state. Any solutions?

**Comment by LazareffR** [comment]
Looking for a solution to the problem

**Comment by joaoaccl** [comment]
For anyone who's still looking for a solution, set the "Do not disturb" mode to on/true in the UI or send a {"do_not_disturb": true} command. Device power will default to off after a power outage.
